pkg/registry/common/expire/nse_server_test.go

```diff
@@ -241,13 +241,56 @@ func TestExpireNSEServer_RefreshFailure(t *testing.T) {
 
 	clockMock.Add(expireTimeout)
 	require.Eventually(t, func() bool {
-		stream, err := c.Find(ctx, &registry.NetworkServiceEndpointQuery{
-			NetworkServiceEndpoint: new(registry.NetworkServiceEndpoint),
-		})
-		require.NoError(t, err)
+		nses, err := find(ctx, c)
+		return err == nil && len(nses) == 0
+	}, testWait, testTick)
+}
+
+func TestExpireNSEServer_UnregisterFailure(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	clockMock := clockmock.New(ctx)
+	ctx = clock.WithClock(ctx, clockMock)
+
+	mem := memory.NewNetworkServiceEndpointRegistryServer()
+
+	s := next.NewNetworkServiceEndpointRegistryServer(
+		serialize.NewNetworkServiceEndpointRegistryServer(),
+		expire.NewNetworkServiceEndpointRegistryServer(ctx, expireTimeout),
+		new(remoteNSEServer), // <-- GRPC invocation
+		mem,
+	)
 
-		_, err = stream.Recv()
-		return err == io.EOF
+	nse, err := s.Register(ctx, &registry.NetworkServiceEndpoint{
+		Name: nseName,
+	})
+	require.NoError(t, err)
+
+	c := adapters.NetworkServiceEndpointServerToClient(mem)
+
+	nses, err := find(ctx, c)
+	require.NoError(t, err)
+	require.Len(t, nses, 1)
+	require.Equal(t, nseName, nses[0].Name)
+
+	unregisterCtx, cancelUnregister := context.WithCancel(ctx)
+	cancelUnregister()
```

Comment:
I'm not sure if this is reliable enough solution.
Someone in the future can add a context check into serialize or expire itself, then Unregister below wouldn't reach expire server, so the test would silently stop verifying correct behavior.
Or someone can modify the test, adding a new element into chain that checks context.

Maybe, we can create an element like `injecterror`.

Reply:
Yep, fixed.